src/Renderer/WaypointListRenderer.cpp

```diff
@@ -89,7 +89,15 @@ Draw(Canvas &canvas, PixelRect rc,
   row_renderer.DrawSecondRow(canvas, rc, buffer);
 
   // Draw waypoint name
-  row_renderer.DrawFirstRow(canvas, rc, waypoint.name.c_str());
+  if (!waypoint.shortname.empty()) {
+    std::string waypoint_title = waypoint.name.c_str();
+    waypoint_title.append(" (",2);
+    waypoint_title.append(waypoint.shortname);
+    waypoint_title.append(")",1);
```

Comment:
This is just as inefficient as before (see https://github.com/XCSoar/XCSoar/pull/219#discussion_r475553548).